integration/tests/tracer/test_tracer_debug_trace_complex_transactions.py

```diff
@@ -171,6 +172,41 @@ def test_trace_iterative_tx_block_timestamp(self, block_timestamp_contract, json
         assert response["result"]["type"] == "CALL"
         assert "error" not in response["result"]
 
+    def test_trace_iterative_tx_block_timestamp_struct_logger_and_call_trace(
+        self, block_timestamp_contract, json_rpc_client
+    ):
+        contract, _ = block_timestamp_contract
+        sender_account = self.accounts[0]
+
+        tx = self.web3_client.make_raw_tx(sender_account)
+        instruction_tx = contract.functions.callIterativeTrx().build_transaction(tx)
+        receipt = self.web3_client.send_transaction(sender_account, instruction_tx)
+        assert receipt["status"] == 1
+        assert self.web3_client.is_trx_iterative(receipt["transactionHash"].hex())
+
+        json_rpc_client.send_rpc(method="eth_getBlockByHash", params=[receipt["blockHash"].hex(), False])
+        tx_data = self.web3_client.get_transaction_by_hash(receipt["transactionHash"].hex())
+
+        check_tracer_struct_log(self.tracer_api, tx_data)
+        check_call_tracer_type(self.tracer_api, tx_data)
+
+    def test_trace_block_timestamp_in_scheduled_tx(self, block_timestamp_contract, json_rpc_client, web3_client_sol):
+        contract, _ = block_timestamp_contract
+        sender_account = self.accounts[0]
+
+        tx = self.web3_client.make_raw_tx(sender_account)
+        instruction_tx = contract.functions.logTimestamp().build_transaction(tx)
+        receipt = self.web3_client.send_transaction(sender_account, instruction_tx)
+        response = json_rpc_client.send_rpc(method="eth_getBlockByHash", params=[receipt["blockHash"].hex(), False])
+        tx_block_timestamp = EthGetBlockByHashResult(**response).result.timestamp
+
+        event_logs = contract.events.Result().process_receipt(receipt)
+        assert event_logs[0]["args"]["block_timestamp"] <= int(tx_block_timestamp, 16)
```

Comment:
let's disscuss this check separately 
it is not what was needed
contract.events.Result().process_receipt(receipt) - is is not tracer call, you check proxy here, receipt values